controllers/trainingjob/trainingjob_controller_test.go

```diff
@@ -227,6 +234,14 @@ var _ = Describe("Reconciling a TrainingJob that exists", func() {
 					createdRequest := req.(*sagemaker.CreateTrainingJobInput)
 					Expect(*createdRequest.TrainingJobName).To(Equal("training-job-name"))
 				})
+
+				It("Does not modify the job name in the spec", func() {
+					ExpectTrainingJobNameInSpec("training-job-name", trainingJob)
```

Comment:
nit: The "training-job-name" should be kept in a constant local to this `Context` block.